src/connection.js

```diff
@@ -1462,74 +1344,29 @@ Connection.prototype.STATE = {
         return this.transitionTo(this.STATE.FINAL);
       },
       message: function() {
-        return this.processLogin7Response();
-      }
-    }
-  },
-  SENT_LOGIN7_WITH_NTLM: {
-    name: 'SentLogin7WithNTLMLogin',
-    events: {
-      socketError: function() {
-        return this.transitionTo(this.STATE.FINAL);
-      },
-      connectTimeout: function() {
-        return this.transitionTo(this.STATE.FINAL);
-      },
-      data: function(data) {
-        if (this.sspiClientResponsePending) {
-          // We got data from the server while we're waiting for getNextBlob()
-          // call to complete on the client. We cannot process server data
-          // until this call completes as the state can change on completion of
-          // the call. Queue it for later.
-          const boundDispatchEvent = this.dispatchEvent.bind(this);
-          return setImmediate(boundDispatchEvent, 'data', data);
-        } else {
-          return this.sendDataToTokenStreamParser(data);
+        // Use SSPI blob if received
+        if (this.ntlmpacketBuffer) {
+          return this.authProvider.processChallenge(this.ntlmpacketBuffer, (error, responseBuffer) => {
```

Comment:
Function call should be `this.authProvider.parseChallengeMessage( )`, authentication provider implementation uses the name `parseChallengeMessage`  😄